src/main/java/com/yourcodereview/jegors/task1/Main.java

```diff
@@ -38,21 +42,20 @@ public static void main(String[] args) {
         var startTime = Instant.now();
         LOGGER.log(INFO, "Execution start time: {0}", startTime);
 
-        process(path);
+        try (var ips = Files.lines(path)) {
+            System.out.println(countUnique(ips));
+        } catch (IOException e) {
+            LOGGER.log(ERROR, "Error during processing file: {0}", path);
+        }
 
         var executionTime = Duration.between(startTime, Instant.now());
         LOGGER.log(INFO, "Execution time: {0}", executionTime);
     }
 
-    private static void process(Path path) {
-        try (var lines = Files.lines(path)) {
-
-            var unique = lines.collect(countingUnique());
-
-            System.out.println(unique);
-
-        } catch (IOException e) {
-            LOGGER.log(ERROR, "Error during processing file: {0}", path);
-        }
+    private static long countUnique(Stream<String> ipAddresses) {
+        return ipAddresses
+                .mapToInt(new IPv4Converter())
```

Comment:
Did you create new IPV4Converter for each handling address? I think it's overdesigned  hear, just replace your converter with simple util method and use static reference.